packages/devtools_app/lib/src/trees.dart

```diff
@@ -138,18 +142,73 @@ class TreeNode<T extends TreeNode<T>> {
     );
   }
 
-  T firstNodeAtLevel(int level) {
+  /// Locates the first sub-node in the tree at level [level].
+  ///
+  /// [level] is relative to the subtree root [this].
+  ///
+  /// For example:
+  ///
+  /// [level 0]                A
+  ///                        /   \
+  /// [level 1]             B     E
+  ///                      /    /  \
+  /// [level 2]           C    F    G
+  ///                    /
+  /// [level 3]         D
+  ///
+  /// E.firstSubNodeAtLevel(1) => F
```

Comment:
Nice diagram in the comment!
Also document what the runtime of this method is in terms of the depth and total size of the tree. That would help to figure out when code using this is slow or not.

I would also recommend filing a bug to add a performance benchmark to track what the total processing time if for a large tree so we make sure we don't end up making it a performance bottleneck for large traces.

Reply:
done